### PR TITLE
ninja stuff

### PR DESCRIPTION
**Pull request by SonixApache**
Ninjas now spawn in the first iteration of the dojo. I hope someone makes it look better some time.
dojo_v2
There's two walls and two airlocks in the dojo to practice sword-cutting on.
There's also some sake and sushi on the table.
On the right, there's an area to practice the glove battery drain function.
Two buttons up top replace the old system of launching; 
- The button marked [SPACE] launches you at the station
- The button marked [SOMETHING IN CHINESE] teleports you into maint.

Ninjas now get a classic ninja energy blade. 
It's invisible when inactive, and doesn't make any noise when activated.

sword_inhand
sword_itemstates

Also moved most of the ninja shit to its own two files.
Also gave the ninjas bigger oxygen tanks.
Also changed the ninja spawn sound into a YOOOOOO as per someone's request.
Ninjas now get put in a faction, so if by any chance more than one spawns, they can identify eachother.
Removed the [Throw ninja from space?) option from the post-bus role application, since it isn't needed anymore and doesn't work anymore anyway.
Added cushions (which are just barstools without metal), an arrow decal, a teleporter decal, and a sign.
Also did some other stuff, probably.

- [x] [UPDATE WIKI]

🆑 
- experiment:  Added a first iteration of the Spider Clan dojo - it contains two airlocks and walls to slice, a punching bag, a teleporter button and a launcher button. And some sake.
- rscadd:  Added a spider clan faction file that handles the dojo and other future stuff
- rscadd:  Moved most of the stuff over to the Ninja and Spider Clan faction files
- imageadd:  Changed the energy katana to the energy blade of the ninjas of yore
- soundadd:  Added a new sound effect for ninjas spawning
- rscdel: Disabled latejoin ninjas. For now.

### CODE REVIEW

**Comment by kernikov** [comment]
3/10
There's like, 16 different sushi food items and you used approximately none of them for the buffet table in your dojo.

File: code/datums/gamemode/dynamic/dynamic_rulesets_midround.dm [hunk 1]
```diff
@@ -455,11 +455,9 @@
 /datum/dynamic_ruleset/midround/from_ghosts/ninja/setup_role(var/datum/role/newninja)
 	newninja.OnPostSetup()
 	newninja.Greet(GREET_MIDROUND)
-	newninja.ForgeObjectives()
-	newninja.AnnounceObjectives()
-	if(!newninja.antag.current.ThrowAtStation())
-		newninja.antag.current.spawn_rand_maintenance()
-	return 1
+	var/datum/faction/spider_clan/spoiderclan = find_active_faction_by_type(/datum/faction/spider_clan)
+	spoiderclan.forgeObjectives()
```
Comment (Kurfursten) [inline code/datums/gamemode/dynamic/dynamic_rulesets_midround.dm]: this is going to make new factionwide objectives on every ninja spawn, so for example if midround ninja fires twice, they'll have a fuckton of faction goals

File: code/datums/gamemode/factions/spider_clan.dm [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+		if(T.z != STATION_Z)
+			continue
+		if(M.stat != DEAD)
+			living++
+	if(living<=16 && prob(25))
+		AppendObjective(/datum/objective/silence)
+	else
+		AppendObjective(/datum/objective/survive)
+	if(prob(15))
+		AppendObjective(/datum/objective/stealsake)
```
Comment (Kurfursten) [inline code/datums/gamemode/factions/spider_clan.dm]: didn't you put sake in the dojo

File: code/game/area/Space Station 13 areas.dm [hunk 3]
```diff
@@ -170,7 +170,13 @@ proc/process_adminbus_teleport_locs()
 /area/no_ethereal
 	anti_ethereal = 1
 
-
+/area/dojo
+	name = "\improper Spider Clan Dojo"
+	icon_state = "dojo"
+	requires_power = 0
```
Comment (Kurfursten) [inline code/game/area/Space Station 13 areas.dm:176]: Might want an area where they can practice draining APCs

**Comment by Kurfursten** [comment]
the dojo should have more things with cells (like a borg, mecha, egun) that you can click to drain power from as practice, maybe some signs on walls to help teach those skills too

File: code/datums/gamemode/dynamic/dynamic_rulesets_midround.dm (hunk 1 above)
Comment (SonixApache) [reply code/datums/gamemode/dynamic/dynamic_rulesets_midround.dm]: good point

File: code/datums/gamemode/factions/spider_clan.dm (hunk 2 above)
Comment (SonixApache) [reply code/datums/gamemode/factions/spider_clan.dm]: yes but they have to survive an entire round with it AND sake heals ninjas

File: code/game/area/Space Station 13 areas.dm (hunk 3 above)
Comment (SonixApache) [reply code/game/area/Space Station 13 areas.dm:176]: definitely among the things i want to add, but not yet

**Comment by SonixApache** [comment]
I agree, but I'd have to make them 'holographic' or otherwise useless. 
Something for future additions; right now I just want to get the basic functionality in.

**Pull request merged**